libraries/chain/chain_controller.cpp

```diff
@@ -248,13 +249,13 @@ bool chain_controller::_push_block(const signed_block& new_block)
  * queues.
  */
 transaction_trace chain_controller::push_transaction(const packed_transaction& trx, uint32_t skip)
-{ try {
-   return with_skip_flags(skip, [&]() {
-      return _db.with_write_lock([&]() {
-         return _push_transaction(trx);
-      });
-   });
-} FC_CAPTURE_AND_RETHROW() }
+   { try {
+         return with_skip_flags(skip, [&]() {
+            return _db.with_write_lock([&]() {
+               return _push_transaction(trx);
+            });
+         });
+      } EOS_CAPTURE_AND_RETHROW( transaction_exception ) }
```

Comment:
`FC_CAPTURE_AND_RETHROW` will catch `std::exception` and `...` and turn them into `fc::exception`. `EOS_CAPTURE_AND_RETHROW` only catches `fc::exception`. I think we should enhance `EOS_CAPTURE_AND_RETHROW` to also catch  `std::exception` and `...` and turn them into `fc::exception`.

Reply:
... which is exactly what you did. That is what I get for writing a comment before fully reviewing.